pkg/ebpf/c/tracee.bpf.c

```diff
@@ -2997,15 +2998,39 @@ int BPF_KPROBE(trace_security_bpf)
     if (should_submit(SECURITY_BPF, p.event)) {
         // 1st argument == cmd (int)
         save_to_submit_buf(p.event, (void *) &cmd, sizeof(int), 0);
-
         events_perf_submit(&p, SECURITY_BPF, 0);
     }
-
     union bpf_attr *attr = (union bpf_attr *) PT_REGS_PARM2(ctx);
 
     reset_event_args(&p);
     check_bpf_link(&p, attr, cmd);
 
+    // Capture BPF object loaded
+    if (cmd == BPF_PROG_LOAD && p.config->options & OPT_CAPTURE_BPF) {
+        bin_args_t bin_args = {};
+        u32 pid = p.task_info->context.host_pid;
+
+        u32 insn_cnt = get_attr_insn_cnt(attr);
+        const struct bpf_insn *insns = get_attr_insns(attr);
+        unsigned int insn_size = (unsigned int) (sizeof(struct bpf_insn) * insn_cnt);
```

Comment:
@AsafEitani FTR only... We don't have a problem on using `sizeof(struct bpf_insn)` here because the `struct bpf_insn` won't ever have its size changed.

BUT, the correct approach for any other kernel types, whenever you want to get its size, is to use the bpf type helper `bpf_core_type_size(type)` for CO-RE and sizeof() for NON-CO-RE. This is major because the type might change from one kernel to another and for portability, the helper will consider that (So you don't need to declare the FULL type in the vmlinux.h file, only the types you really use).